sound/soc/sof/intel/hda-dsp.c

```diff
@@ -334,8 +334,9 @@ static int hda_dsp_send_pm_gate_ipc(struct snd_sof_dev *sdev, u32 flags)
 	pm_gate.flags = flags;
 
 	/* send pm_gate ipc to dsp */
-	return sof_ipc_tx_message(sdev->ipc, pm_gate.hdr.cmd, &pm_gate,
-				  sizeof(pm_gate), &reply, sizeof(reply));
+	return sof_ipc_tx_message_nopm(sdev->ipc, pm_gate.hdr.cmd,
```

Comment:
is this change really necessary? You'd always send the pm_date_ipc before the DSp is put in D0i3, isnt it?

Reply:
It is needed, without this x_nopm() helpers, it will wake up the DSP to D0I0 before sending the pm_gate_ipc.